File: game_prototype/Statuses.py

import math

class Statuses():
    rage = 0
    poison = 0
    burn = 0
    frozen = 0
    frightened = 0
    confused = 0
    paralyzed = 0
    clear = []
    
    def clearStatuses(self):
        self.rage = 0
        self.poison = 0
        self.burn = 0
        self.frozen = 0
        self.frightened = 0
        self.confused = 0
        self.paralyzed = 0


    def addStatus(self, status, length):
        self.setStatus(status, 1)
        self.clear.append([status, length])
        
    def setStatus(self, status, value):
        if status == "rage":
            self.rage = value
        elif status == "poison":
            self.poison = value
        elif status == "burn":
            self.burn = value
        elif status == "frozen":
            self.frozen = value
        elif status == "frightened":
            self.frightened = value
        elif status == "confused":
            self.confused = value
        elif status == "paralyzed":
            self.paralyzed = value
        else:
            print("Invalid status")


    def calculateAttack(self, attack):
        
        if self.rage:
            attack = math.ceil(attack * 1.3)
        if self.frightened:
            attack = math.ceil(attack * .75)
        
        return attack

    def checkStatus(self, time):
        tempList = []
        for i in self.clear:
            if i[1] <= time:
                self.clear.remove(i)
                self.setStatus(i[0], 0)
                tempList.append("Cleared: " + str(i[0]) + "\n")
                 
        return tempList
    

if __name__ == "__main__":
    status = Statuses()
    status.addStatus("rage", 10)
    status.checkStatus(5)
    status.checkStatus(15)

File: frontend/TodoFromScratch/app/src/main/java/com/example/todofromscratch/model/domain/Statuses.java

package com.example.todofromscratch.model.domain;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

public class Statuses {
    private int rage;
    private int poison;
    private int burn;
    private int frozen;
    private int frightened;
    private int confused;
    private int paralyzed;
    private List<String[]> clear;

    public Statuses() {
        rage = 0;
        poison = 0;
        burn = 0;
        frozen = 0;
        frightened = 0;
        confused = 0;
        paralyzed = 0;
        clear = new ArrayList<>();
    }

    public void clearStatuses() {
        rage = 0;
        poison = 0;
        burn = 0;
        frozen = 0;
        frightened = 0;
        confused = 0;
        paralyzed = 0;
    }

    public void addStatus(String status, int length) {
        setStatus(status, length);
        clear.add(new String[]{status, Integer.toString(length)});
    }

    public void setStatus(String status, int value) {
        switch (status) {
            case "rage":
                rage = value;
                break;
            case "poison":
                poison = value;
                break;
            case "burn":
                burn = value;
                break;
            case "frozen":
                frozen = value;
                break;
            case "frightened":
                frightened = value;
                break;
            case "confused":
                confused = value;
                break;
            case "paralyzed":
                paralyzed = value;
                break;
            default:
                System.out.println("Invalid status");
        }
    }

    public int calculateAttack(int attack) {
        if (rage != 0) {
            attack = (int) Math.ceil(attack * 1.3);
        }
        if (frightened != 0) {
            attack = (int) Math.ceil(attack * 0.75);
        }
        return attack;
    }

    public String getStatusString(){
        StringBuilder tempStr = new StringBuilder();
        if (rage != 0) {
            tempStr.append("Rage: ").append(rage).append(" ");
        }
        if (poison != 0) {
            tempStr.append("Poison: ").append(poison).append(" ");
        }
        if (burn != 0) {
            tempStr.append("Burn: ").append(burn).append(" ");
        }
        if (frozen != 0) {
            tempStr.append("Frozen: ").append(frozen).append(" ");
        }
        if (frightened != 0) {
            tempStr.append("Frightened: ").append(frightened).append(" ");
        }
        if (confused != 0) {
            tempStr.append("Confused: ").append(confused).append(" ");
        }
        if (paralyzed != 0) {
            tempStr.append("Paralyzed: ").append(paralyzed).append(" ");
        }
        return tempStr.toString();
    }

    public List<String> checkStatus(int time) {
        List<String> tempList = new ArrayList<>();
        Iterator<String[]> iterator = clear.iterator();

        while (iterator.hasNext()) {
            String[] i = iterator.next();
            if (Integer.parseInt(i[1]) == 0) {
                iterator.remove();
                setStatus(i[0], 0);
                tempList.add("Cleared status: " + i[0]);
            }
            else if(Integer.parseInt(i[1]) > 0) {
                setStatus(i[0], Integer.parseInt(i[1]) - 1);
                //Change the value in clear
                i[1] = Integer.toString(Integer.parseInt(i[1]) - 1);
            }


        }
        return tempList;
    }

    public boolean isRage() {
        return rage!=0;
    }

    public boolean isFrightened() {
        return frightened!=0;
    }
    public boolean isConfused() {
        return confused!=0;
    }
    public static void main(String[] args) {
        Statuses status = new Statuses();
        status.addStatus("rage", 10);
        status.checkStatus(5);
        System.out.print(status.checkStatus(15));
    }
}
